hypothesisfuzzer/fuzzing/repo_fuzzer.py

```diff
@@ -16,74 +17,126 @@
+
         self.name = name
         self.config = config
         self._clone_git(config['git_url'])
         self._create_venv()
         self._start_fuzzing()
 
+        self.log.debug('Repository fuzzer initialised.')
+
     def on_webhook(self, payload):
+
+        self.log.debug('Webhook event began.')
+
         # Check if repo is the same name as the one set in config
+        payload_name = payload["repository"]["name"]
+        config_name = self.config['repo_name']
         try:
-            if payload["repository"]["name"] != self.config['repo_name']:
+            if payload_name != config_name:
+                self.log.warning('Repository %s is different from %s.',
+                                 payload_name,
+                                 config_name)
                 return 'OK'
         except KeyError:
+            self.log.info('Repository %s is as configured.', payload_name)
             pass
 
         try:
+            self.log.debug('Cloning repository %s.', payload_name)
             self._clone_git(self.config['git_url'])
+            self.log.debug('Repository %s cloned.', payload_name)
         except Exception:
-            return generic_error(msg="Error cloning Git Repo! " +
+            self.log.error('Unable to access repository %s for cloning.',
+                           payload_name)
+            return generic_error(msg="Error cloning Git repository! " +
                                      "Please ensure you have access.")
 
         self._stop_fuzzing()
+        self.log.debug('Old repository fuzzing stopped.')
         self._create_venv()
+        self.log.debug('Virtual environment created and fuzzing restarted.')
         self._start_fuzzing()
 
+        self.log.debug('Webhook event ended.')
+
         return 'OK'
 
     def get_commit_hash(self):
 
+        self.log.debug('Getting commit hash for repository %s.', self.name)
+
         if not os.path.exists(self.name):
+            self.log.error('When getting commit hash, path of %s not found.',
+                           self.name)
             return no_code_dir_error()
         repo = GitRepo(self.name)
         sha = repo.head.object.hexsha
 
+        self.log.debug('Commit hash for repository %s obtained.', self.name)
+
         return jsonify({
             "sha": sha
         })
 
     def get_errors(self):
+
+        self.log.debug('Getting errors for repository %s.', self.name)
+
         if not os.path.exists(self.name):
+            self.log.error('When getting errors, path of %s not found.',
+                           self.name)
             return no_code_dir_error()
         with open(self.name+'/data.txt', 'r') as file_data:
+            self.log.debug('Errors for repository %s obtained.', self.name)
             return jsonify(json.load(file_data))
 
     def _clone_git(self, git_url):
-        # Delete old code folder
 
+        self.log.debug('Deleting repository %s.', self.name)
```

Comment:
Move this inside the if statement, because it won't delete anything if its not there.

Reply:
Would also add 'old' repository, not just repository.